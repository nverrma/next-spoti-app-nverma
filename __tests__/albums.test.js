import { render, screen } from "@testing-library/react";
import Home from "@/pages/albums";

describe("albums component testing", () => {
  const albums = [
    {
      id: 1,
      image: "https://i.scdn.co/image/ab67616d0000b27358406b3f1ac3ceaff7a64fef",
      name: "dash",
    },
    {
      id: 2,
      image: "https://i.scdn.co/image/ab67616d0000b27358406b3f1ac3ceaff7a64fef",
      name: "dash",
    },
  ];

  it("component to be true", () => {
    const data = render(<Home />);
    expect(data).toBeTruthy();
  });

  it("component to be true", () => {
    const data = render(<Home />);
    expect(data.findByText("Playlist")).toBeTruthy();
  });
  
  it("component to be true", () => {
    expect(albums).toEqual(
      expect.arrayContaining([
        expect.objectContaining({ id: 1 }),
        expect.objectContaining({ id: 2 }),
      ])
    );
  });
});
