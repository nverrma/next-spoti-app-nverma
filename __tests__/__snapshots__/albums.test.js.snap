// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`albums renders a heading 1`] = `
<div>
  <div
    style="background: skyblue;"
  >
    <h1
      class="text-center my-5"
    >
      Playlist
    </h1>
    <div
      class="row"
    />
    <h1
      class="my-5 text-center"
    >
      your album playlist
    </h1>
    <div
      class="row"
    />
  </div>
</div>
`;
